proto/rill/runtime/v1/resources.proto

```diff
@@ -641,6 +641,7 @@ message CanvasSpec {
   uint32 gap = 3;
   repeated ComponentVariable variables = 5;
   repeated CanvasItem items = 4;
+  repeated SecurityRule security_rules = 6;
```

Comment:
should this be supported on component level or canvas level ? Seems like component will have mapping with either model or mv so that makes security mapping more appropriate ?

Reply:
It may also be needed on `Component`s at some point, but the main goal here is to support blocking access to the end dashboards for folks they are not relevant for or that you know may not have access to the data inside them.

Queries made from components are still subject to the underlying API or metrics view's security policies, so not having security policies on the component itself does not leak data. See the PR comment above:

> Among other use cases, this enables hiding dashboards that will anyway fail to render for people who don't have access to the underlying metrics views.
> 
> A policy on an explore/canvas resource does not prevent direct queries to the underlying metrics views, so it is still recommended to define security policies directly on metrics views.

Reply:
I was more thinking on lines if canvas has multiple components accessing different mv and user has security policy on few and not all.

Reply:
Okay that makes sense but would also involve some work on the canvas implementation side, so will discuss and implement that idea separately